### ISSUE TITLE
Rewrite search highlighting so it can use css

### ISSUE THREAD

**Issue by friday** [Ulauncher/Ulauncher#1177]
### Description

Currently themes use a css file an a manifest json file. The latter is almost completely useless.

```json
{
    "manifest_version": "1",
    "name": "dark",
    "display_name": "Elementary Dark",
    "extend_theme": "light",
    "css_file": "theme.css",
    "matched_text_hl_colors": {
        "when_selected": "#fff",
        "when_not_selected": "#fff"
    }
}
```

It contains information we don't need like "name" (just use the css file name), but it has two things we do need: `extend_theme` for inheritance and `matched_text_hl_colors` for highlighting.

The former can be solved in other ways (including reset.css for all themes so they don't have to inherit and/or adding a special css import syntax)

The `matched_text_hl_colors` are needed because of an implementation detail. The result uses a regular css label with "pango html" support, and we're adding spans. This can't be styled with css. We should instead switch to using a horizontal Gtk Box with multiple labels as children.

This way themes could use pure css and be easier to write.

### Communication guidelines

- [X] I have read and agree with Ulauncher's communication guidelines and checked the [existing enhancement requests](https://github.com/Ulauncher/Ulauncher/labels/enhancement) before suggesting this.

**Comment by Laurens256** [comment]
I *almost* got this working.The only part of this I don't see being possible is how to ellipsize the text if it gets too long. Since there would be multiple labels, and there's no ellipsize attribute for `Gtk.box`, the text doesn't ellipsize correctly and the container could potentially get super wide for long titles

https://github.com/Laurens256/Ulauncher/tree/highlighting-rewrite

**Comment by friday** [comment]
You can make a draft PR if you want @Laurens256. That way makes it easier to share, test and collaborate :).

I did test it though and it worked very well.

I didn't consider the ellipsize problem when I created the issue. It can't be solved in the same way as before, but I think if the unhighlighted labels ellipsize, then it's an improvement from before. I haven't tested that though. The second (description) row is always longer for me, so usually that's the only one that gets shortened.

If you want to change the abstraction a bit more and have an idea of how you want to do that, also feel free to do that. The new behavior warrants a different abstraction and terminology in the methods imo. It doesn't have to be part of this PR, but if you already have ideas for that definitely don't feel like you have to respect the old method names.

And you are definitely underselling yourself "I'm not really experienced with stuff like this but I'll give it a go", then diving head first into this GTK specific code with the glade xml file, pango documentation, the matching_blocks output format etc... :muscle: :pray:

**Comment by friday** [comment]
Fixed in #1293 (themes still use a json file, but no longer need `matched_text_hl_colors`)
